nova_vm/src/ecmascript/builtins/ordinary.rs

```diff
@@ -957,7 +966,13 @@ pub(crate) fn ordinary_object_create_with_intrinsics(
         ProtoIntrinsics::WeakMap => agent.heap.create(WeakMapHeapData::default()).into_object(),
         ProtoIntrinsics::WeakRef => agent.heap.create(WeakRefHeapData::default()).into_object(),
         ProtoIntrinsics::WeakSet => agent.heap.create(WeakSetHeapData::default()).into_object(),
+    };
+
+    if let Some(prototype) = prototype {
+        object.internal_set_prototype(agent, Some(prototype));
```

Comment:
**praise:** This is such a nice and clean solution, great idea! I'd been thinking that this function would maybe eventually somehow pass the `prototype` to all of the individual heap data constructors and they'd get to decide if they create the backing object immediately or not. But that was a huge hassle which is why I avoided doing it for the time being.

This, however <3 A great, greeeeat idea.